java/flight/flight-sql/src/main/java/org/apache/arrow/flight/sql/FlightSqlProducer.java

```diff
@@ -0,0 +1,720 @@
+    public static final Schema GET_SCHEMAS_SCHEMA = new Schema(
+        Arrays.asList(Field.nullable("catalog_name", MinorType.VARCHAR.getType()),
+            Field.nullable("schema_name", MinorType.VARCHAR.getType())));
+    public static final Schema GET_IMPORTED_AND_EXPORTED_KEYS_SCHEMA = new Schema(Arrays.asList(
+        Field.nullable("pk_catalog_name", MinorType.VARCHAR.getType()),
+        Field.nullable("pk_schema_name", MinorType.VARCHAR.getType()),
+        Field.nullable("pk_table_name", MinorType.VARCHAR.getType()),
+        Field.nullable("pk_column_name", MinorType.VARCHAR.getType()),
+        Field.nullable("fk_catalog_name", MinorType.VARCHAR.getType()),
+        Field.nullable("fk_schema_name", MinorType.VARCHAR.getType()),
+        Field.nullable("fk_table_name", MinorType.VARCHAR.getType()),
+        Field.nullable("fk_column_name", MinorType.VARCHAR.getType()),
+        Field.nullable("key_sequence", MinorType.INT.getType()),
+        Field.nullable("fk_key_name", MinorType.VARCHAR.getType()),
+        Field.nullable("pk_key_name", MinorType.VARCHAR.getType()),
+        Field.nullable("update_rule", new ArrowType.Int(8, false)),
+        Field.nullable("delete_rule", new ArrowType.Int(8, false))));
+    public static final Schema GET_SQL_INFO_SCHEMA =
+        new Schema(Arrays.asList(
+            Field.nullable("info_name", new ArrowType.Int(32, false)),
+            new Field("value",
+                // dense_union<string_value: string, int_value: int32, bigint_value: int64, int32_bitmask: int32>
+                new FieldType(true, new Union(UnionMode.Dense, new int[] {0, 1, 2, 3}), /*dictionary=*/null),
+                Arrays.asList(
+                    Field.nullable("string_value", MinorType.VARCHAR.getType()),
+                    Field.nullable("int_value", MinorType.INT.getType()),
+                    Field.nullable("bigint_value", MinorType.BIGINT.getType()),
+                    Field.nullable("int32_bitmask", MinorType.INT.getType())))));
+
+    private Schemas() {
+      // Prevent instantiation.
+    }
+  }
+
+  /**
+   * Reserved options for the SQL command `GetSqlInfo` used by {@link FlightSqlProducer}.
+   */
+  final class SqlInfo {
+    public static final int FLIGHT_SQL_SERVER_NAME = 0;
+    public static final int FLIGHT_SQL_SERVER_VERSION = 1;
+    public static final int FLIGHT_SQL_SERVER_ARROW_VERSION = 2;
+    public static final int FLIGHT_SQL_SERVER_READ_ONLY = 3;
+    public static final int SQL_DDL_CATALOG = 500;
+    public static final int SQL_DDL_SCHEMA = 501;
+    public static final int SQL_DDL_TABLE = 502;
+    public static final int SQL_IDENTIFIER_CASE = 503;
+    public static final int SQL_IDENTIFIER_QUOTE_CHAR = 504;
+    public static final int SQL_QUOTED_IDENTIFIER_CASE = 505;
+  }
+
+  /**
+   * Update/delete rules for {@link FlightSqlProducer#getStreamImportedKeys} and
+   * {@link FlightSqlProducer#getStreamExportedKeys}.
+   */
+  final class UpdateDeleteRules {
```

Comment:
If I'm not mistaken, this isn't used anywhere?

If it is to be used, might it make more sense as an enum? (If it's to appear in an API anywhere.)

Reply:
This is supposed to be used when implementing FlightSqlProducer#getStreamImportedKeys and getStreamExportedKeys, the client should also use this when using the response of these flights.

Reply:
Ah right, got it.